app/common/notifications/notify.service.js

```diff
@@ -217,7 +218,52 @@ function Notify(_, $q, $rootScope, $translate, SliderService, ModalService, Demo
 
         return deferred.promise;
     }
+    function deleteWithInput (type, checkName) {
+        var scope = getScope();
+        var deferred = $q.defer();
+        scope.noequal = false;
+
+        const modalText = {
+            survey: {
+                title: 'notify.form.delete_form_confirm',
+                modalBody: 'notify.form.delete_form_confirm_desc',
+                errorLabel: 'notify.form.delete_form_error',
+                button: 'notify.form.delete_form_button'
+            }
+        };
+
+        var texts = modalText[type];
 
+        $translate(texts.modalBody, {check_name: checkName}).then(show, show);
+
+        function show(body) {
+
+            scope.confirm = function (name) {
+                if (name === checkName) {
```

Comment:
may look a tiny bit cleaner if we do

scope.isEqual = name === checkName; 
and then 
if (!scope.isEqual) {do blah}

defaulting to false for isEqual as you do of course.

Reply:
Totally! Thanks!